Fixes for August 2017 new index

shelleydoljack opened the pull request:
@ndushay, a PR for you to merge.

ndushay commented on spec/advanced_search_spec.rb:
```diff
@@ -131,15 +131,15 @@
       end
       it 'subject as a phrase' do
         expect(@sub_phrase.size).to be >= 500
```
Consider pumping up the minimum also - maybe 550?

shelleydoljack replied:
ok.

ndushay commented on spec/advanced_search_spec.rb:
```diff
@@ -131,15 +131,15 @@
-        # resp.should have_the_same_number_of_results_as(solr_resp_ids_from_query('Socialization'))
-        expect(resp.size).to be >= 475_000
-        expect(resp.size).to be <= 525_000
+        expect(resp).to have_the_same_number_of_results_as(solr_resp_ids_from_query('Socialization'))
```
if you leave this line commented out, does it work?

shelleydoljack replied:
Yes, test will pass with line 140 commented out.

ndushay commented on spec/advanced_search_spec.rb:
```diff
@@ -402,8 +402,14 @@
         resp = solr_resp_doc_ids_only({ 'fq' => 'format:("Video"), language:("English"), building_facet:("Green"), topic_facet:("Feature films")', 'q' => 'collection:*' }.merge(solr_args))
-        expect(resp.size).to be >= 25
-        expect(resp.size).to be <= 100
+        # 655a was taken out of topic facet, hence smaller number of records expected.
```
this comment will become outdated quickly.  It's helpful to me ... but perhaps it should be a comment here in the PR, rather than in the code?  What do you think?

ndushay commented on spec/advanced_search_spec.rb:
```diff
@@ -416,13 +422,24 @@
         resp = solr_resp_doc_ids_only({ 'fq' => 'format:("Video"), language:("English"), building_facet:("Media & Microtext Center"), topic_facet:("Feature films")', 'q' => 'collection:*' }.merge(solr_args))
-        expect(resp.size).to be >= 10_500
-        expect(resp.size).to be <= 21_000
+        # 655a was taken out of topic facet, hence smaller number of records expected.
```
ditto on this comment

ndushay commented on spec/boolean_spec.rb:
```diff
@@ -225,7 +225,8 @@
           # titles 'Indochine'
-          indochine_results = %w(383033 430603 4312384 3083716 3065221 2134301)
+          indochine_results = %w(383033 430603 1921469 3083716 3065221 2134301 10316540)
+          # 4312384 withdrawn
```
ditto on this comment - another way to deal with this is to make an individual commit with this in the commit message (maybe you did - I haven't looked at individual commits yet)

ndushay commented on spec/boolean_spec.rb:
```diff
@@ -255,17 +256,21 @@
+    context 'street art and graffiti', jira: 'VUF-1013', fixme: true do
       it '((street art) OR graffiti) AND aspects' do
+        # this search returns 47 results when the solr query is sent from SearchWorks
+        # "q"=>"( ( _query_:\"{!edismax qf=$qf_subject pf=$pf_subject pf3=$pf3_subject pf2=$pf2_subject}street art\" OR _query_:\"{!dismax qf=$qf_subject pf=$pf_subject pf3=$pf3_subject pf2=$pf2_subject}graffiti\" ) AND _query_:\"{!dismax qf=$qf_subject pf=$pf_subject pf3=$pf3_subject pf2=$pf2_subject}aspects\" )"
```
perhaps we can work together to change this to be the query actually sent by solr.  I'm thinking of spinning up SW locally, pointing to a particular Solr instance, and then tailing the logs to see what the query sent was ...

ndushay commented on spec/boolean_spec.rb:
```diff
@@ -255,17 +256,21 @@
         expect(resp.size).to be <= 4070
       end
       it '("street art" OR graffiti) AND aspects' do
+        # this search returns 39 results when the solr query is sent from SearchWorks
```
ditto for above

ndushay commented on spec/cjk/chinese_author_spec.rb:
```diff
@@ -44,14 +44,14 @@
   end
 
-  context "Zhang, Ailing", :jira => 'SW-207' do
+  context "Zhang, Ailing", :jira => 'SW-207', fixme: true do
```
maybe we can work together to figure this out and get it working

ndushay commented on spec/cjk/chinese_zhengzhou_geography_spec.rb:
```diff
@@ -86,7 +86,7 @@
       end
       # there are a LOT more of these --> socrates finds 1953 of them.
-      it "should rank high the documents with one word in 245a" do
+      it "should rank high the documents with one word in 245a", fixme: true do
```
another test we might pair on to try to keep working

ndushay commented on spec/cjk/korean_author_spec.rb:
```diff
@@ -47,13 +48,13 @@
-      in_245c_and_700 = '7850201'  # 245c has [著者 강 인철
-      it_behaves_like 'best matches first', 'everything', query, [in_245c, in_245c_and_700], 13
+      in_245c_and_700 = '11987524'  # 245c has 편집 강 인철.
+      it_behaves_like 'best matches first', 'everything', query, [in_245c, in_245c_and_700], 15
```
great job!

ndushay commented on spec/cjk/korean_spacing_spec.rb:
```diff
@@ -45,12 +45,14 @@
         # it_behaves_like 'good results for query', 'everything', '한미 동맹', 1, 25, '8375648', 20
-        it_behaves_like "expected result size", 'everything', '한미 동맹', 1, 25
+        # the characters are entered as "한, 미 동맹" in 245a, maybe the comma is messing something up?
+        # .880. 00|6245-01|a21세기 의 한, 미 동맹 관계 /|c이 춘근 편.
```
another candidate for pairing, perhaps?

ndushay commented on spec/cjk/korean_spacing_spec.rb:
```diff
@@ -102,7 +104,8 @@
       end
-      context "꿈꾸는 자가 창조한 다 (but Korean writing would not give space between 한 and 다)" do
+      context "꿈꾸는 자가 창조한 다 (but Korean writing would not give space between 한 and 다)", fixme: true do
+        # FIXME: ckey 7881507 shows up, not sure why (can't find these chars in the record)
```
pairing?

ndushay commented on spec/cjk/korean_spacing_spec.rb:
```diff
@@ -261,8 +264,9 @@
-  context "New writings on understanding of contemporary North Korea" do
+  context "New writings on understanding of contemporary North Korea", fixme: true do
     shared_examples_for "good results for 새로 쓴 현대북한의 이해" do | query |
+      # FIXME: ckey 7881507 shows up, not sure why (can't find these chars in the record)
```
Pairing?

ndushay commented on spec/journal_title_spec.rb:
```diff
@@ -8,7 +8,7 @@
       orig_query_str = solr_params['q'].split('}').last
       resp = solr_resp_ids_titles(solr_params)
-      expect(resp).to include({'title_245a_display' => /^#{orig_query_str}\W*$/i}).in_each_of_first(exp_ids.size)
+      expect(resp).to include({'title_245a_display' => /^#{orig_query_str}.*$/i}).in_each_of_first(exp_ids.size)
```
this is no longer "exact title";  `\W` means NOT a-z, A-Z, 0-9, or _ (underscore)  Another one for pairing?

ndushay commented on spec/journal_title_spec.rb:
```diff
@@ -290,7 +290,7 @@
   end # the state
 
-  context "the world" do
+  context "the world", fixme: true do
```
pairing?

ndushay commented on spec/journal_title_spec.rb:
```diff
@@ -471,7 +474,7 @@
   end
 
-  context "the atlantic" do
+  context "the atlantic", fixme: true do
```
pairing?

ndushay commented on spec/subject_search_spec.rb:
```diff
@@ -230,7 +230,8 @@
       resp = solr_resp_doc_ids_only(subject_search_args('geology').merge(fq: 'format:Database'))
       expect(resp.size).to be >= 10
-      expect(resp).to include('3982587')
+      # ckey 3982587 withdrawn
```
comment has no meaning outside this PR

shelleydoljack commented:
@ndushay I think this is ready for merging now.

ndushay commented on spec/advanced_search_spec.rb:
```diff
@@ -130,16 +130,16 @@
-        # resp.should have_the_same_number_of_results_as(solr_resp_ids_from_query('Socialization'))
-        expect(resp.size).to be >= 475_000
-        expect(resp.size).to be <= 525_000
+        expect(resp).to have_fewer_results_than(solr_resp_ids_from_query('Socialization'))
```
I no longer remember why this is true -- that keyword in advanced search would have fewer results than plain everything search.  But I trust you.

shelleydoljack replied:
The solr query sent for advanced search uses defType=lucene, hence fewer results.

ndushay replied:
ah yes - should there be a comment to that effect?

ndushay commented on spec/dismax_mm_setting_spec.rb:
```diff
@@ -28,6 +28,12 @@
+      resp = solr_resp_doc_ids_only(subject_search_args 'France Social life and customs 20th century')
+      expect(resp.size).to be <= 500
+      expect(resp).to have_fewer_documents_than(solr_resp_doc_ids_only(subject_search_args 'France Social life customs century'))
+    end
```
excellent adding this spec for the mm change!

ndushay commented on spec/journal_title_spec.rb:
```diff
@@ -291,7 +291,7 @@
 
   context "the world" do
-    it_behaves_like "great results for format journal", "The world" do
+    it_behaves_like "great results for format journal", "The world", :fixme => true do
```
I no longer remember - did we give up on "The world" journal?  Or did you just forget to remove the fixme?

shelleydoljack replied:
I kept the fixme because the linked 880's containing the parallel titles are legitimate in cataloging and affecting the results. I'm not sure there is a way to fix the test (or how we index the records).

ndushay replied:
oh yeah!  Now I remember.

ndushay replied:
should this be in the comments somewhere?

ndushay commented on spec/journal_title_spec.rb:
```diff
@@ -445,7 +448,7 @@
 
   context "the journal" do
-    it_behaves_like "great results for journal/newspaper", "The journal" do
+    it_behaves_like "great results for journal/newspaper", "The journal", :fixme => true do
```
I no longer remember - did we give up on "The journal"?  Or did you just forget to remove the fixme?

shelleydoljack replied:
Same as above.

ndushay commented:
I also just remembered to look at the commit messages.   I wouldn't hold up this PR for it, but it's generally good for commit messages to help someone looking at commit history hone in on exactly what was done.  

a76948c6 has a good short commit message "Advanced search returns fewer records than normal search."  but what about "advanced_search_spec: keyword returns fewer records than normal search due to diff defTypes"  

3aabf9ad089 "Tests fail because parallel titles in linked 880 fields skewing relevancy" - almost a great message -- but it says nothing about WHICH tests are failing, and what the fix is.  Maybe "journal_title_spec: 2 tests now fixme due to linked 880 fields skewing relevancy" would be better?

These are less good:

fb1b697103 - "Marks tests as fixme's." -- not very useful.  Even "temporarily mark failing tests as fixmes" is more informative.   Or having a separate commit for each file changed with the filename in it -- anything that would help you if you needed to look at commit history and figure out what happened.

bc0bd9e94f758c - "Ckey 7881507 is part of the results now." - almost.  If you prefix it with "korean_spacing" it makes a lot more sense, IMO.

etc.

I frequently do a significant rebasing of my commits before my PR is ready for merging.   If you want some help, I'd be happy to show you what I know.

shelleydoljack commented:
@ndushay ok, I think I have rebased it correctly. For one of the generic commits, I had split it into more specific, detail commits but neglected to drop the generic commit. I think I've fixed that now.

ndushay commented:
Hi Shelley,

You have my 👍 to merge this yourself.  I believe I suggested maybe adding comments to a couple specs above (advanced_search, journal_title), but if you don't want to, that's fine.

- Naomi
